impeller/tessellator/tessellator.cc

```diff
@@ -468,6 +468,34 @@ EllipticalVertexGenerator Tessellator::FilledEllipse(
                                    });
 }
 
+EllipticalVertexGenerator Tessellator::FilledRoundRect(
+    const Matrix& view_transform,
+    const Rect& bounds,
+    const Size& radii) {
+  if (radii.width * 2 < bounds.GetSize().width &&
+      radii.height * 2 < bounds.GetSize().height) {
+    auto max_radius = radii.MaxDimension();
+    auto divisions = ComputeQuadrantDivisions(
+        view_transform.GetMaxBasisLength() * max_radius);
+    auto upper_left = bounds.GetLeftTop() + radii;
+    auto lower_right = bounds.GetRightBottom() - radii;
+    return EllipticalVertexGenerator(Tessellator::GenerateFilledRoundRect,
+                                     GetTrigsForDivisions(divisions),
+                                     PrimitiveType::kTriangleStrip, 4,
+                                     {
+                                         .reference_centers =
+                                             {
+                                                 upper_left,
+                                                 lower_right,
+                                             },
+                                         .radii = radii,
+                                         .half_width = -1.0f,
+                                     });
+  } else {
+    return FilledEllipse(view_transform, bounds);
+  }
```

Comment:
uber-nit

```suggestion
  }
  return FilledEllipse(view_transform, bounds);
```

Reply:
I'm curious what is the reasoning for the nit.

One advantage of using the else is that it makes sure that the "previous implementation code" did, in fact, return - otherwise the method would lack a return statement.

I just ran into a case of that within this PR where I forgot to return after "doing one version of the work" of a method and so ended up doing "both versions of the work" instead.

Reply:
It's basically a case of "do this OR do that".

Also, if it is "do 1 line of code or do 30 lines of code", readability might be improved by "if { 1 line; return }; 30 lines" but in this case both blocks are small.

Reply:
I think that is a fair point. Generally I would say that I prefer that we use as little nesting as possible. So if you think of this as an early return, then it makes sense to remove the nesting. As you point out, you could also think about this as being, return one of two options, so it makes sense to leave the else.

Some thoughts on the matter can be seen at https://engdoc.corp.google.com/eng/doc/tott/episodes/651.md?cl=head and https://engdoc.corp.google.com/eng/doc/tott/episodes/680.md?cl=head

I'd say taking either approach is fine, I tend to (overly simplistically probably) push for unnesting.